integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/APIKeyTestCase.java

```diff
@@ -119,11 +126,24 @@ public void invokeAPIKeyInHeaderSuccessTest() throws Exception {
                 "Response code mismatched");
     }
 
+    //Invoke API by including the API key as a query parameter
+    @Test(description = "Test to check the API Key in query param is working")
+    public void invokeAPIKeyInQueryParamSuccessTest() throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        HttpResponse response = HttpClientRequest.doGet(
+                Utils.getServiceURLHttps(endPoint + "?apikey=" + apiKey), headers);
+
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(),
+                com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus.SC_OK,
+                "Response code mismatched");
+    }
+
     @Test(description = "Test to check the API key auth validate invalid signature key")
     public void invokeAPIKeyHeaderInvalidTokenTest() throws Exception {
         // Set header
         Map<String, String> headers = new HashMap<>();
-        headers.put("x-api-key", TestConstant.INVALID_JWT_TOKEN);
```

Comment:
What if we want to enable API key from APIM but use a custom query param or header name for API key. Is that not possible?

Reply:
no, it's not possible. as apim only allows api key name to be apikey. we can't use swagger security components in apim scenario. https://apim.docs.wso2.com/en/latest/design/api-security/api-authentication/secure-apis-using-api-keys/